literate_notebooks/src-ES/13_extras.jl

```diff
@@ -1,175 +1,179 @@
-freqtable(x) # order is preserved and not-used level is shown
+levels!(x, [3, 1, 2, 4]) # reorganizar niveles y añadir un nivel extra
+freqtable(x) # el orden se manteine y los nivelos no usados se meustran
 
 #-
 
-freqtable([1,1,2,3,missing]) # by default missings are listed
+freqtable([1,1,2,3,missing]) # por default los valores faltatnes (`missing`s) se muestran en lista
 
 #-
 
-freqtable([1,1,2,3,missing], skipmissing=true) # but we can skip them
+freqtable([1,1,2,3,missing], skipmissing=true) # pero nos los podemos saltar
 
-# ### DataFramesMeta - working on `DataFrame`
+# ### DataFramesMeta - trabajando en un `DataFrame`
 
 using DataFramesMeta
 df = DataFrame(x=1:8, y='a':'h', z=repeat([true,false], outer=4))
 
 #-
 
-@with(df, :x+:z) # expressions with columns of DataFrame
+@with(df, :x+:z) # expresiones con columnas de DataFrame
 
 #-
 
-@with df begin # you can define code blocks
+@with df begin # puedes definir bloques de código
     a = :x[:z]
     b = :x[.!:z]
     :y + [a; b]
 end
 
 #-
 
-a # @with creates hard scope so variables do not leak out
+a # `@with` crea un  `hard scope` (alcance léxico fuerte) y así las variables no se derraman al ambiente
 
 #-
 
 df2 = DataFrame(a = [:a, :b, :c])
-@with(df2, :a .== ^(:a)) # sometimes we want to work on raw Symbol, ^() escapes it
+@with(df2, :a .== ^(:a)) # A veces queremos un `Symbol` a secas, usamos ^() para escapar esa secuencia
 
 #-
 
 df2 = DataFrame(x=1:3, y=4:6, z=7:9)
-@with(df2, _I_(2:3)) # _I_(expression) is translated to df2[expression]
+@with(df2, _I_(2:3)) #   `_I_(expresión)` se traduce a `df2[expression]`
 
 #-
 
-@where(df, :x .< 4, :z .== true) # very useful macro for filtering
+@where(df, :x .< 4, :z .== true) # macro muy útil para filtrar
 
 #-
 
-@select(df, :x, y = 2*:x, z=:y) # create a new DataFrame based on the old one
+@select(df, :x, y = 2*:x, z=:y) # crea un nuevo DataFrame basado en el anterior
 
 #-
 
-@transform(df, a=1, x = 2*:x, y=:x) # create a new DataFrame adding columns based on the old one
+@transform(df, a=1, x = 2*:x, y=:x) # crea un DataFrame agregando columnas basado en el anterior
 
 #-
 
-@transform(df, a=1, b=:a) # old DataFrame is used and :a is not present there
+@transform(df, a=1, b=:a) # se usa el DataFrame anterior y `:a` no está presente ahí
 
 #-
 
-@orderby(df, :z, -:x) # sorting into a new data frame, less powerful than sort, but lightweight
+@orderby(df, :z, -:x) # ordenando los datos en un nuevo DataFrame, menos poderoso que `sort` pero mucho más ligero
 
 #-
 
-@linq df |> # chaining of operations on DataFrame
+@linq df |> # podemos encadenar operaciones sobre un DataFrame
     where(:x .< 5) |>
     orderby(:z) |>
     transform(x²=:x.^2) |>
     select(:z, :x, :x²)
 
 #-
 
-f(df, col) = df[col] # you can define your own functions and put them in the chain
+f(df, col) = df[col] # puees definir tus propias funcioens y ponerlas en una cadena (`chain` = cadena, como las de metal)
```

Comment:
puedes, instead of puees

Reply:
mira, ando inventando palabras. Gracias!